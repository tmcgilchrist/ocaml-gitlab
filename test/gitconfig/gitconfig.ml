open Gitconfig

let print_config fmt config =
  let open Format in
  let rec print_bindings fmt = function
  | [] -> ()
  | (key, value) :: rem ->
      fprintf fmt "\t%s = %s\n" key value;
      print_bindings fmt rem
  in
  config |> List.iter (fun (name, bindings) ->
      fprintf fmt "[%s]\n%a" name print_bindings bindings
    )


let () =
  let lexbuf = Lexing.from_string {|[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = git@github.com:tmcgilchrist/ocaml-gitlab
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[branch "parse_gitconfig"]
	remote = origin
	merge = refs/heads/parse_gitconfig
|}
  in
  try
    let config = Gitconfig_parser.config Gitconfig_lexer.token lexbuf in
    print_config Format.std_formatter config
  with
  | Gitconfig_lexer.Error msg ->
      Printf.fprintf stderr "%s%!" msg
  | Gitconfig_parser.Error ->
      Printf.fprintf stderr "At offset %d: syntax error.\n%!" (Lexing.lexeme_start lexbuf)
