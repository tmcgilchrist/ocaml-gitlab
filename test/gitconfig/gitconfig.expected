[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = git@github.com:tmcgilchrist/ocaml-gitlab
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[branch "parse_gitconfig"]
	remote = origin
	merge = refs/heads/parse_gitconfig
